stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/handler/internal/EventHandlerSequential.kt

```diff
@@ -0,0 +1,797 @@
+        when (event) {
+            is ReactionNewEvent -> event.reaction.messageId
+            is ReactionDeletedEvent -> event.reaction.messageId
+            is MessageDeletedEvent -> event.message.id
+            is MessageUpdatedEvent -> event.message.id
+            is NewMessageEvent -> event.message.id
+            is NotificationMessageNewEvent -> event.message.id
+            is ReactionUpdateEvent -> event.message.id
+            else -> null
+        }
+    }
+
+    private fun StateFlow<List<Member>>.containsWithUserId(userId: String): Boolean {
+        return value.find { it.user.id == userId } != null
+    }
+
+    /**
+     * Checks if unread counts should be updated for particular channel.
+     * The unread counts should not be updated if channel in the DB
+     * does not contain [ChannelCapabilities.READ_EVENTS] capability.
+     *
+     * @param cid CID of the channel.
+     *
+     * @return True if unread counts should be updated
+     */
+    private suspend fun RepositoryFacade.hasReadEventsCapability(cid: String): Boolean {
+        return selectChannels(listOf(cid)).let { channels ->
+            val channel = channels.firstOrNull()
+            if (channel?.ownCapabilities?.contains(ChannelCapabilities.READ_EVENTS) == true) {
+                true
+            } else {
+                logger.d {
+                    "Skipping unread counts update for channel: $cid. ${ChannelCapabilities.READ_EVENTS} capability is missing."
+                }
+                false
+            }
+        }
+    }
+
+    private fun Message.enrichWithOwnReactions(batch: EventBatchUpdate, currentUserId: UserId, eventUser: User?) {
+        ownReactions = if (eventUser != null && currentUserId != eventUser.id) {
+            batch.getCurrentMessage(id)?.ownReactions ?: mutableListOf()
+        } else {
+            mergeReactions(
+                latestReactions.filter { it.userId == currentUserId },
+                batch.getCurrentMessage(id)?.ownReactions ?: mutableListOf()
+            ).toMutableList()
+        }
+    }
+
+    private fun GlobalMutableState.updateCurrentUser(me: User) {
+        _user.value = me
+        _mutedUsers.value = me.mutes
+        _channelMutes.value = me.channelMutes
+        _totalUnreadCount.value = me.totalUnreadCount
+        _channelUnreadCount.value = me.unreadChannels
+        _banned.value = me.banned
+    }
+
+    private infix fun UserId.mustBe(currentUserId: UserId?) {
+        if (this != currentUserId) {
+            throw InputMismatchException(
+                "received connect event for user with id $this while for user configured " +
+                    "has id $currentUserId. Looks like there's a problem in the user set"
+            )
+        }
+    }
+
+    companion object {
+        val EMPTY_DISPOSABLE = object : Disposable {
+            override val isDisposed: Boolean = true
+            override fun dispose() {}
```

Comment:
```suggestion
            override fun dispose() = Unit
```
this should remove Detekt warning